nodes/_base/node_selection.py

```diff
@@ -1,22 +1,30 @@
 import bpy
 
+from bpy.props import EnumProperty
 from bpy.types import Node
 from .._base.node_base import ScNode
 from ...helper import focus_on_object
 
 class ScSelectionNode(ScNode):
+
+    in_selection_type: EnumProperty(name="Mode", items=[("VERT", "Vertices", "", "VERTEXSEL", 1), ("EDGE", "Edges", "", "EDGESEL", 2), ("FACE", "Faces", "", "FACESEL", 4)], default=set(), options={"ENUM_FLAG"}, update=ScNode.update_value)
+
     def init(self, context):
         self.node_executable = True
         super().init(context)
         self.inputs.new("ScNodeSocketObject", "Object")
+        self.inputs.new("ScNodeSocketSelectionType", "Selection Type").init("in_selection_type")
         self.outputs.new("ScNodeSocketObject", "Object")
     
     def error_condition(self):
         return (
             self.inputs["Object"].default_value == None
```

Comment:
add a condition to check whether the set is empty

Reply:
As I said in the pull request: 
> By default selection nodes have no selection type specified, and use the current behavior of using the edit mode selection type.

This allows the default behavior to be the same as it currently is.